neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -806,26 +806,43 @@ object Neo4jStoreSingleton : GraphStore {
-                            val regionId = (regionCode ?: map[simProfile.id])
-                                    ?: ValidationError(type = simProfileEntity.name, id = simProfile.iccId, message = "SimProfile not linked to any region")
-                                            .left()
-                                            .bind()
-
-                            val simEntry = simManager.getSimProfile(
-                                    hlr = hssNameLookup.getHssName(regionId),
-                                    iccId = simProfile.iccId)
-                                    .mapLeft { NotFoundError(type = simProfileEntity.name, id = simProfile.iccId) }
-                                    .bind()
-                            org.ostelco.prime.model.SimProfile(
-                                    iccId = simProfile.iccId,
-                                    alias = simProfile.alias,
-                                    eSimActivationCode = simEntry.eSimActivationCode,
-                                    status = simEntry.status)
-                        }
-                    }.fix()
+            Either.monad<StoreError>().binding {
+                simProfiles.bind().map { simProfile ->
+
+                    val regionId = (regionCode ?: map[simProfile.id])
+
+                    val simEntry = if (regionId != null) {
+                        simManager.getSimProfile(
+                                hlr = hssNameLookup.getHssName(regionId),
+                                iccId = simProfile.iccId).fold(
+                                { error ->
+                                    logger.warn("SimProfile not found in SIM Manager DB. region: {}, iccId: {}, error: {}", regionId, simProfile.iccId, error)
+                                    SimEntry(
+                                            iccId = simProfile.iccId,
+                                            status = NOT_READY,
+                                            eSimActivationCode = "Dummy eSIM",
+                                            msisdnList = emptyList()
+                                    )
+                                },
+                                { it }
+                        )
+                    } else {
+                        logger.warn("SimProfile not linked to any region. iccId: {}", simProfile.iccId)
+                        SimEntry(
+                                iccId = simProfile.iccId,
+                                status = NOT_READY,
+                                eSimActivationCode = "Dummy eSIM",
+                                msisdnList = emptyList()
+                        )
+                    }
+
+                    org.ostelco.prime.model.SimProfile(
+                            iccId = simProfile.iccId,
+                            alias = simProfile.alias,
+                            eSimActivationCode = simEntry.eSimActivationCode,
+                            status = simEntry.status)
```

Comment:
Not so sure about `status`. The SIM DB has the field 'provisionstate' which keep track of the state of the SIM (AVAILABLE, RESERVED, PROVISIONED, ALLOCATION_FAILED) and `simManager.getSimProfile()` ignores  this value I think. Depends on context whether this is OK or not...

Reply:
A SIM entry in the DB has three state fields:
   - hlrstate
   - smdplusstate
   - provisonstate
and all of them must be taken into account if you want the actual state of  a SIM entry.

Reply:
But no problem with the change as such.

Reply:
All these state fields listed above are internal to SIM Manager.
They all are mapped to simplified status externally.
The status in modified code is that external status and not the internal status.

Reply:
Yes, only `provisionstate` of internal states is used and is mapped to external states.
This is done here - https://github.com/ostelco/ostelco-core/blob/develop/sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SimManager.kt#L48-L59

Reply:
Whether other states should be used in this mapping implementation details of SIM Manager.
External status is what other modules see.
SIM manager may change the mapping logic without affecting other modules.
But this is not in the scope of this PR.